toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -186,18 +189,70 @@ func getBuildDependencies(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, g
 	return
 }
 
+// parseCheckSection reads the package build log file to determine if the %check section passed or not
+func parseCheckSection(logFile string) (err error) {
+	file, err := os.Open(logFile)
+	// If we can't open the log file, that's a build error.
+	if err != nil {
+		logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logFile, err)
+		return
+	}
+	defer file.Close()
+	for scanner := bufio.NewScanner(file); scanner.Scan(); {
+		currLine := scanner.Text()
+		// Anything besides 0 is a failed test
+		if strings.Contains(currLine, "CHECK DONE") {
+			if strings.Contains(currLine, "EXIT STATUS 0") {
+				return
+			}
+			failedLogFile := strings.TrimSuffix(logFile, ".log")
+			failedLogFile = fmt.Sprintf("%s-FAILED_TEST-%d.log", failedLogFile, time.Now().UnixMilli())
+			err = os.Rename(logFile, failedLogFile)
+			if err != nil {
+				logger.Log.Errorf("Log file rename failed. Error: %v", err)
+				return
+			}
+			err = fmt.Errorf("package test failed. Test status line: %s", currLine)
+			return
+		}
+	}
+	return
+}
+
 // buildSRPMFile sends an SRPM to a build agent to build.
-func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, srpmFile, outArch string, dependencies []string) (builtFiles []string, logFile string, err error) {
+func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, checkAttempts int, srpmFile, outArch string, dependencies []string) (builtFiles []string, logFile string, err error) {
 	const (
 		retryDuration = time.Second
 	)
 
+	// checkFailed is a flag to see if a non-null buildErr is from the %check section
+	checkFailed := false
 	logBaseName := filepath.Base(srpmFile) + ".log"
+	// temporary solution; potential fix: build normally for buildAttempts, then run rmpbuild -bi --short-circuit to just do the checks
```

Comment:
Thanks for the comment. Might be worth to also add a link to the bug/task tracking this work.